parser.go

```diff
@@ -705,11 +687,11 @@ invalid:
 
 // parseSimpleSelectorSequence parses a selector sequence that applies to
 // a single element.
-func (p *parser) parseSimpleSelectorSequence() (Selector, error) {
-	var result Selector
+func (p *parser) parseSimpleSelectorSequence() (compoundSelector, error) {
+	var result compoundSelector
```

Comment:
The return type here should be Matcher rather than compoundSelector, so that if there is only one selector in the sequence, you can return that selector directly instead of wrapping it in a compoundSelector with only one child.